### PR TITLE
added files for RPM packaging

### PR DESCRIPTION
**Pull request by simonc6372**
Added the support files for packaging into an RPM, and a short script to build the RPM.

### CODE REVIEW

File: packaging/make_package [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  -n grafana-agent \
+  --license "Apache 2.0" \
+  --vendor "Grafana Labs" \
+  --url "https://github.com/grafana/agent" \
+  --after-install after_install.sh \
+  --before-remove before_remove.sh \
+  --config-files etc/grafana-agent.yaml \
+  --config-files etc/sysconfig/grafana-agent  \
+  ../dist/agent-linux-amd64=/usr/bin/grafana-agent \
+  var=/ grafana-agent.service=/usr/lib/systemd/system/grafana-agent.service
```
Comment (jdbaldry) [inline packaging/make_package:18]:
```suggestion
  grafana-agent.service=/usr/lib/systemd/system/grafana-agent.service
```
I think var=/ needs to be removed as there is no var directory when I tested locally.

File: packaging/make_package [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/bash
+#
+#we want to name the binary grafana-agent not agent-linux-amd64
```
Comment (jdbaldry) [inline packaging/make_package:3]:
```suggestion
# Build an RPM package from the linux binary.
# This script assumes the binary has already been built in "${GIT_ROOT}"/dist, the script is being run from the "${GIT_ROOT}"/packaging directory, and that we want to name the binary grafana-agent not agent-linux-amd64.
```

File: packaging/make_package (hunk 1 above)
Comment (jdbaldry) [reply packaging/make_package:18]: In fact, I think the var is missing from the repository. When I try and test the package with the centos-systemd image in [] I see an error relating to the `var` directory:

```
$ # From the packaging directory
$ docker run -it --tmpfs /run --tmpfs /run/lock -v /sys/fs/cgroup:/sys/fs/cgroup:ro -v "$(git rev-parse --show-toplevel)"/dist:/dist quay.io/cortexproject/centos-systemd:latest rpm -i /dist/grafana-agent-0.4.0-1.x86_64.rpm

chown: cannot access '/var/lib/grafana-agent': No such file or directory
warning: %post(grafana-agent-0.4.0-1.x86_64) scriptlet failed, exit status 1
```

File: packaging/etc/grafana-agent.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+server:    #configuration options for the embeded server it self. 
+  http_listen_address: '127.0.0.1'  # local host only. 
+  http_listen_port: 9090  
+  
+prometheus:
+  global:
+    scrape_interval:     15s  # By default, scrape targets every 15 seconds.
+    external_labels:          # Static labels to add to all metrics
+      scanned_by: 'Grafana Agent' 
+      scanner_on: 'local.snc.me.uk'
```
[the remark this answers is not in the document]
Comment (simonc6372) [reply packaging/etc/grafana-agent.yaml:11]: that makes sense. also changed the name to something in example.com

**Comment by tomwilkie** [comment]
Cool, didn't realise we had this - would be good to get this done and start publishing packages!

**Comment by rfratto** [comment]
Closing in favor of #200, but thanks for all the work that was done on this!

**Pull request closed without merging**